site/docs/relations/logical_relations.md

```diff
@@ -310,6 +310,37 @@ If at least one grouping expression is present, the aggregation is allowed to no
 %%% proto.algebra.AggregateRel %%%
     ```
 
+## Reference Operator
+
+The reference operator is used to construct DAGs of operations. In a `Plan` we can have multiple Rel representing various 
+computations with potentially multiple outputs. The `ReferenceRel` is used to express the fact that multiple `Rel` might be
+sharing subtrees of computation. This can be used to express arbitrary DAGs as well as represent multi-query optimizations.
+
+As a concrete example think about two queries `SELECT * FROM A JOIN B JOIN C` and `SELECT * FROM A JOIN B JOIN D`,
+We could use the `ReferenceRel` to highlight the shared `A JOIN B` between the two queries, by creating a plan with 3 `Rel`. 
+One expressing `A JOIN B` (in position 0 in the plan), one using reference as follows: `ReferenceRel(0) JOIN C` and a third one
+doing `ReferenceRel(0) JOIN D`. This allows to avoid the redundancy of `A JOIN B`.
```

Comment:
I'm confused by this example because it seems to imply we can have multiple queries expressed in a single plan.  What is the root of the DAG here?  I think this is two DAGs.  I thought a reference rel would be more related to subqueries.